api/types/apicache.go

```diff
@@ -468,7 +468,7 @@ type blockHeap []*CachedBlock
 
 // BlockPriorityQueue implements heap.Interface and holds CachedBlocks
 type BlockPriorityQueue struct {
-	*sync.RWMutex
+	mtx                  sync.RWMutex
```

Comment:
Must be a pointer because the Sort interface copies the mutex.

Reply:
changed func instead.

Reply:
Changing to pointer receivers works too.  Is this fundamentally better though?